lib/matplotlib/axes/_axes.py

```diff
@@ -1122,15 +1122,19 @@ def hlines(self, y, xmin, xmax, colors='k', linestyles='solid',
         if not np.iterable(xmax):
             xmax = [xmax]
 
-        y, xmin, xmax = cbook.delete_masked_points(y, xmin, xmax)
-
+        # Create and combine masked_arrays from input
+        y, xmin, xmax = cbook._combine_masks(y, xmin, xmax)
         y = np.ravel(y)
-        xmin = np.resize(xmin, y.shape)
-        xmax = np.resize(xmax, y.shape)
+        xmin = np.ravel(xmin)
+        xmax = np.ravel(xmax)
+
+        masked_verts = np.ma.empty((len(y), 2, 2))
+        masked_verts[:, 0, 1] = y
+        masked_verts[:, 1, 1] = y
+        masked_verts[:, 0, 0] = xmin
+        masked_verts[:, 1, 0] = xmax
```

Comment:
Can you put these in order? It makes more sense to me this way.
```suggestion
        masked_verts[:, 0, 0] = xmin
        masked_verts[:, 0, 1] = y
        masked_verts[:, 1, 0] = xmax
        masked_verts[:, 1, 1] = y
```